trie/triedb/hashdb/database.go

```diff
@@ -736,7 +736,6 @@ func (db *Database) Size() (common.StorageSize, common.StorageSize) {
 func (db *Database) Close() error {
 	if db.cleans != nil {
 		db.cleans.Reset()
-		db.cleans = nil
```

Comment:
why did we remove this?

Reply:
It causes a race in testing. Applied to upstream as well: https://github.com/ethereum/go-ethereum/pull/29309